qir/qat/DivisionByZeroPass/DivisionByZeroPass.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "qir/qat/DivisionByZeroPass/DivisionByZeroPass.hpp"
+
+#include "qir/qat/Llvm/Llvm.hpp"
+#include "qir/qat/Logging/ILogger.hpp"
+#include "qir/qat/QatTypes/QatTypes.hpp"
+
+#include <functional>
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft::quantum
+{
+char* const   DivisionByZeroPass::EC_REPORT_FUNCTION      = "__qir__report_error_value";
+char* const   DivisionByZeroPass::EC_VARIABLE_NAME        = "__qir__error_code";
```

Comment:
Currently it is const pointer to non-const data (while the data is the string literal residing in the read-only data segment).
Consider disabling the update of the string literal:
```suggestion
const char* const   DivisionByZeroPass::EC_REPORT_FUNCTION      = "__qir__report_error_value";
const char* const   DivisionByZeroPass::EC_VARIABLE_NAME        = "__qir__error_code";
```